e2e/test/iothub/DeviceTokenRefreshE2ETests.cs

```diff
@@ -35,11 +34,20 @@ public async Task IotHubDeviceClient_Not_Exist_AMQP()
             using var deviceClient = IotHubDeviceClient.CreateFromConnectionString(
                 $"HostName={config.IotHubHostName};DeviceId=device_id_not_exist;SharedAccessKey={config.SharedAccessKey}",
                 options);
-            await deviceClient.OpenAsync().ConfigureAwait(false);
+
+            try
+            {
+                await deviceClient.OpenAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                // assert
+                Assert.AreEqual<Type>(ex.GetType(), typeof(IotHubClientException));
+                Assert.AreEqual<IotHubStatusCode>(((IotHubClientException)ex).StatusCode, IotHubStatusCode.DeviceNotFound);
+            }
```

Comment:
Instead of using `ExpectedException` attributes, I am using try-catch here to check for both exception type & status code, any thoughts for this? @drwill-ms

Reply:
https://fluentassertions.com/exceptions/